apps/hub/lib/hub.ex

```diff
@@ -0,0 +1,34 @@
+defmodule Hub do
```

Comment:
I am starting to smell `delegate`. Will this module do anything more? Like, why cannot this be done directly in `subscriber`?
```elixir
    Enum.each(@publishers, fn %{name: name, init_arg: _, module: publisher} ->
      publisher.handle_message(name, message)
    end)
```
The communication will always be `subscribers -> publishers` and never other way around, right?

Reply:
True. My thinking is that hub does the spread out of a single message to all publishers. However the motivation of Hub server is the assumption that sending messages between servers is relatively hard operation. Thats why I wanted to make it a bit easier for `Handler`s of subscribers. Instead of copying a messages to all publishers, they would just send a message to Hub and not care anymore.

But I am starting to think that this assumption might not be correct. I will read about it a bit. If it turns out that it is incorrect then I will move this logic to `Handler`. But if it holds, then IMO it is valid point.

Reply:
idk, `Handler` already has the message, it does not copy. It, it would just call (asynchronnaly, without waiting) M `publishers` instead of calling one `Hub`.